### DISCUSSION TITLE
Function to check GGUF version?

### DISCUSSION THREAD

**Discussion by danemadsen** [ggml-org/llama.cpp#3955] [Q&A] [answered]
Is there a function i can use to check which GGUF version a model is on without hard loading it with something like llama_init_from_gpt_params. Im wanting to do this so i can tell users their model is too old in my app without relying on error handling.

**Comment by KerfuffleV2** [comment, marked as answer]
It's kind of hacky but you can just check the 5th byte in the file. The field is a `uint32` but this should work until after GGUF version 255.

For example:

    00000000  47 47 55 46 02 00 00 00  ed 00                    |GGUF......|

That's a GGUFv2 file.

    00000000  47 47 55 46 03 00 00 00  1f 02                    |GGUF......|

That's a GGUFv3 file.
